OpenIGTLinkIF/Command/vtkSlicerOpenIGTLinkIFCommand.cxx

```diff
@@ -0,0 +1,301 @@
+    os << "None" << "\n";
+    }
+  os << "DeviceId: " << this->GetDeviceID() << "\n";
+  os << "CommandId: " << this->GetQueryID() << "\n";
+}
+
+//----------------------------------------------------------------------------
+const char* vtkSlicerOpenIGTLinkIFCommand::GetCommandName()
+{
+  return this->GetCommandAttribute("Name");
+}
+
+//----------------------------------------------------------------------------
+void vtkSlicerOpenIGTLinkIFCommand::SetCommandName(const char* name)
+{
+  this->SetCommandAttribute("Name", name);
+  this->Modified();
+}
+
+//----------------------------------------------------------------------------
+const char* vtkSlicerOpenIGTLinkIFCommand::GetCommandAttribute(const char* attName)
+{
+  return this->CommandXML->GetAttribute(attName);
+}
+
+//----------------------------------------------------------------------------
+void vtkSlicerOpenIGTLinkIFCommand::SetCommandAttribute(const char* attName, const char* attValue)
+{
+  this->CommandXML->SetAttribute(attName, attValue);
+}
+
+//----------------------------------------------------------------------------
+const char* vtkSlicerOpenIGTLinkIFCommand::GetCommandText()
+{
+  std::ostringstream os;
+  this->CommandXML->PrintXML(os, vtkIndent(0));
+  this->SetExportedCommandText(os.str().c_str());
+  return this->ExportedCommandText;
+}
+
+//----------------------------------------------------------------------------
+bool vtkSlicerOpenIGTLinkIFCommand::SetCommandText(const char* text)
+{
+  if (text==NULL)
+    {
+    vtkErrorMacro("Failed to set OpenIGTLink command from text: empty input");
+    return false;
+    }
+
+  vtkSmartPointer<vtkXMLDataElement> parsedElem = vtkSmartPointer<vtkXMLDataElement>::Take(vtkXMLUtilities::ReadElementFromString(text));
+  if (parsedElem == NULL)
+  {
+    vtkErrorMacro("Failed to set OpenIGTLink command from text: not an XML string: "<<text);
+    return false;
+  }
+
+  this->CommandXML->DeepCopy(parsedElem);
+  return true;
+}
+
+//----------------------------------------------------------------------------
+int vtkSlicerOpenIGTLinkIFCommand::GetNumberOfResponses()
+{
+  int numberOfNestedElements = this->ResponseXML->GetNumberOfNestedElements();
+  return numberOfNestedElements;
+}
+
+//----------------------------------------------------------------------------
+const char* vtkSlicerOpenIGTLinkIFCommand::GetResponseMessage(int responseID/*=0*/)
+{
+  if (this->ResponseXML == nullptr)
+    {
+    return "";
+    }
+
+  if (responseID == 0)
+    {
+    // We might expect the response message to be stored in XML as:
+    // <Command Message="MESSAGE_TEXT"></Command>
+    const char* responseAttribute = this->GetResponseAttribute("Message");
+    if (responseAttribute)
+      {
+      return responseAttribute;
+      }
+    }
+
+  // If responseID is > 0 or if above is not true, look instead for:
+  // <Command><Response Success="true/false" Message="MESSAGE_TEXT"></Response></Command>
+  vtkXMLDataElement* nestedElement;
```

Comment:
vtkXMLDataElement needs to be deleted after usage, otherwise will cause memory leak. 
Call nestedElemetn->Delete() after usage

Reply:
Correct me if I'm wrong, but I don't think that I need to delete the pointer. It's stored and allocated in the root element. This is storing a temporary reference to the nested element.
Trying to delete it results in an error.

Reply:
Hi Kyle,

yes, you are right, i was correcting another memory leak in OpenIGTLinkIO library. and though this was the same case. the pointer is used to access the this->ResponseXML, which is a member of vtkSlicerOpenIGTLinkIFCommand. Apparently you did it correct! Sorry for the confusion.